app/src/androidTest/java/org/simple/clinic/patient/PatientSyncAndroidTest.kt

```diff
@@ -2,122 +2,55 @@ package org.simple.clinic.patient
 
 import android.support.test.runner.AndroidJUnit4
 import com.f2prateek.rx.preferences2.Preference
-import com.google.common.truth.Truth.assertThat
-import io.bloco.faker.Faker
-import io.reactivex.Completable
-import io.reactivex.Observable
-import io.reactivex.Single
 import org.junit.Before
-import org.junit.Rule
-import org.junit.Test
 import org.junit.runner.RunWith
-import org.simple.clinic.AppDatabase
-import org.simple.clinic.AuthenticationRule
 import org.simple.clinic.TestClinicApp
 import org.simple.clinic.TestData
+import org.simple.clinic.patient.sync.PatientPayload
 import org.simple.clinic.patient.sync.PatientPushRequest
-import org.simple.clinic.patient.sync.PatientSync
+import org.simple.clinic.patient.sync.PatientSync2
 import org.simple.clinic.patient.sync.PatientSyncApiV1
-import org.simple.clinic.sync.SyncConfig
-import org.simple.clinic.user.UserSession
-import org.simple.clinic.util.Just
+import org.simple.clinic.sync.BaseDataSyncAndroidTest
 import org.simple.clinic.util.Optional
 import org.threeten.bp.Instant
-import java.text.SimpleDateFormat
 import javax.inject.Inject
 import javax.inject.Named
 
 @RunWith(AndroidJUnit4::class)
-class PatientSyncAndroidTest {
-
-  @Inject
-  lateinit var configProvider: Single<SyncConfig>
+class PatientSyncAndroidTest : BaseDataSyncAndroidTest<PatientSaveModel, PatientPayload>() {
```

Comment:
It might be easier to view this as an individual file. I created a separate `PatientSync2AndroidTest` and wrote it separately, then renamed it `PatientSyncAndroidTest` and rebased it with this commit, but that makes the diff inconvenient to parse.